cmd/archeio/app/handlers.go

```diff
@@ -72,7 +72,12 @@ func MakeHandler(rc RegistryConfig) http.Handler {
 
 func makeV2Handler(rc RegistryConfig, blobs blobChecker) func(w http.ResponseWriter, r *http.Request) {
 	// matches blob requests, captures the requested blob hash
-	reBlob := regexp.MustCompile("^/v2/.*/blobs/sha256:([0-9a-f]{64})$")
+	// https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pull
+	// Blobs are at `/v2/<name>/blobs/<digest>`
+	// Note that ':' cannot be contained in <name> but *must* be contained in <digest>
+	// <digest> also cannot contain `/` so we can use a relatively simple and cheap regex
+	// to match blob requests and capture the digest
+	reBlob := regexp.MustCompile("^/v2/.*/blobs/([^/]+:[a-zA-Z0-9=_-]+)$")
```

Comment:
@BenTheElder this regexp compile happens only once right?

Reply:
Yes, and we're only swapping the existing regex for one that permits and captures arbitrary digest algorithms.